### PR TITLE
chore: refactor Ref to Symbol

### PR DESCRIPTION
**Pull request by wonbyte**
Given the role of `Ref` in handling unique identifiers and ensuring type safety, renaming `Ref` to `Symbol` seems appropriate. The term "symbol" more accurately reflects its purpose and usage in the codebase.

This work addresses #32

### CODE REVIEW

File: riot/runtime/core/proc_effect.ml [hunk 1]
```diff
@@ -2,7 +2,7 @@ open Util
 
 type _ Effect.t +=
   | Receive : {
-      ref : 'a Ref.t option;
+      ref : 'a Symbol.t option;
```
Comment (wonbyte) [inline riot/runtime/core/proc_effect.ml:5]: @leostera I assume we want to change this `ref` to be `symbol` as well? Just want to confirm.

File: riot/lib/ref.ml [hunk 2]
```diff
@@ -1 +1 @@
-include Runtime.Core.Ref
```
Comment (leostera) [inline riot/lib/ref.ml:1]: We should also move this file around and make it `symbol.ml`, so we export this on `riot.mli`.

**Comment by leostera** [comment]
Hi @wonbyte! Thanks for the PR ✨ – it looks good :) we just gotta export this new symbol module through `riot.mli` and fix the docs there.

Btw, the PR description triggered my AI spidey sense – is it AI?

**Comment by wonbyte** [comment]
I did run it through some grammar and spell checking so nice catch. Want to make sure I properly document everything as clearly as possible. Not sure i fully understand yet! I'll make those updates shortly.

**Pull request closed without merging**